lib/ansible/modules/unarchive.py

```diff
@@ -427,7 +427,8 @@ def is_unarchived(self):
             unarchived = False
 
         # Get some information related to user/group ownership
-        umask = os.umask(0)
+        # Loose file permission has been replaced with stricter permission
+        umask = os.umask(0o777)
```

Comment:
that is not what is happening here, we are assigning the 'existing umask' asn resetting it on the next line to what it was, making it 'stricter' is a noop.